src/account_multisig.ts

```diff
@@ -175,48 +176,49 @@ export class Account2FA extends AccountMultisig {
             ...fak2lak.map((pk) => addKey(pk, functionCallAccessKey(accountId, MULTISIG_CHANGE_METHODS, null))),
             addKey(confirmOnlyKey, functionCallAccessKey(accountId, MULTISIG_CONFIRM_METHODS, null)),
             deployContract(contractBytes),
-        ]
+        ];
         if ((await this.state()).code_hash === '11111111111111111111111111111111') {
-            actions.push(functionCall('new', newArgs, MULTISIG_GAS, MULTISIG_DEPOSIT),)
+            actions.push(functionCall('new', newArgs, MULTISIG_GAS, MULTISIG_DEPOSIT),);
         }
-        console.log('deploying multisig contract for', accountId)
+        console.log('deploying multisig contract for', accountId);
         return await super.signAndSendTransactionWithAccount(accountId, actions);
     }
 
     async disable(contractBytes: Uint8Array) {
-        const { accountId } = this
-        const accessKeys = await this.getAccessKeys()
-        const lak2fak = accessKeys.filter(({ access_key }) => 
-            access_key && access_key.permission && access_key.permission.FunctionCall &&
-            access_key.permission.FunctionCall.receiver_id === accountId &&
-            access_key.permission.FunctionCall.method_names &&
-            access_key.permission.FunctionCall.method_names.length === 4 &&
-            access_key.permission.FunctionCall.method_names.includes('add_request_and_confirm')    
-        )
-        const confirmOnlyKey = PublicKey.from((await this.postSignedJson('/2fa/getAccessKey', { accountId })).publicKey)
+        const { accountId } = this;
+        const accessKeys = await this.getAccessKeys();
+        const lak2fak = accessKeys
+            .filter(({ access_key }) => access_key.permission['FunctionCall'] !== undefined)
+            .filter(({ access_key }) => {
+                const perm = (access_key.permission as FunctionCallPermissionView).FunctionCall;
```

Comment:
I think it makes sense to change type for `permission` to define fields like `FunctionCall` so that there is no need to use `['FunctionCall']` and do a cast to `FunctionCallPermissionView`

Reply:
the trouble is that the field `permission` returned by the RPC is either an object or a string. We need to have some check to see which we have.

Could change it to: `typeof access_key.permission !== 'string'`

Reply:
@mehtaphysical got it. Now I understand it's unfortunate format choice on backend, so it's not really possible to avoid explicit cast for `permission`